### ISSUE TITLE
JAWS in Chrome does not read tables correctly when they have rowspans

### ISSUE THREAD

**Issue by gdkraus** [FreedomScientific/standards-support#193]
### Summary

In Chrome, when reading a table with rowspans, reading down a column stops reading before the end of the column is reached.

Example:
1. Go to https://www.w3.org/WAI/tutorials/tables/irregular/
2. Go to the second table example, the one with rowspans
3. Use the table navigation keys go to the first cell with "A2" (Zodiac, Full Color, Sizes Available)
4. Navigate down the column

As you navigate down the column, you will notice that the VC is moving one column to the right each time you go down, until eventually you get to the end of the table and you are told you are at the end of the column, when in fact you are not. 

### Expected result

I should be able to read straight down the column.

### Actual result

JAWS stops reading the column before the end of the actual end of the column.

### Example

See the text above

#### JAWS version and build number

JAWS 2018 and 2019.1903.47
 
#### Operating System and version

Win 10

#### Browser and version:

Chrome 74.0.3729.108

**Comment by gdkraus** [comment]
NVDA in Chrome works.
JAWS in IE 11 works.

**Comment by janellenchew** [comment]
Noting here that in Chrome, the VC also jumps around differently depending on which direction you are moving. 

For example, if I start in "A2" (Zodiac, Full Color), and navigate one cell down, I will be in "A2" (Zodiac, Black and White). Then, if I navigate one cell up, I will be in "A3" (Zodiac, Full Color), instead of "A2" (Zodiac, Full Color) where I originally started.

I can also reproduce this bug in Firefox 68 (Windows 10 and JAWS 2019.1810.64) where JAWS does not read straight down the column, except the VC does not jump around dependent on the direction as I described for Chrome.

For example, if I start in "A2" (Zodiac, Full Color), and navigate one cell down, I will be in "A2" (Zodiac, Black and White). Then, if I navigate one cell up, I will be back in "A2" (Zodiac, Full Color).

**Comment by menovak** [comment]
July 2019
quick update:

Testing with JAWS 2019.1904, Chrome 75 and Firefox 67 
- I was able to reproduce the odd behavior when attempting to navigate using the table screen reading keys, either up or down the columns (under Sizes available).  The resultant navigation path is slightly different in Chrome versus Firefox, but neither is correct 

Testing with NVDA 2019.1, Chrome 75 and Firefox 67  
- using the table screen reading keys, either up or down or across the columns (under Sizes available), the correct table cells (and column/row headers) were announced

Will file this as a JAWS bug

**Comment by ms9513** [comment]
A fix was incorporated as an [enhancement](https://support.freedomscientific.com/Downloads/JAWS/JAWSWhatsNew#Enhancements) in version 2020.2001.70 (February 2002):
"Added the ability for JAWS and Fusion to properly recognize and report both Column Span and Row Span info in HTML tables where information might span over 2 or more cells."

Tested and confirmed fixed.
Pages tested:
- [Example 2: Table with headers spanning multiple rows or columns](https://www.w3.org/WAI/tutorials/tables/irregular/#table-with-headers-spanning-multiple-rows-or-columns)
- [HTML Tables with JAWS](https://www.freedomscientific.com/SurfsUp/Tables.htm) | Table under header "Sales Contact by State"

**Comment by JAWS-test** [comment]
The problem no longer exists in Chrome (with JAWS 2020). The output is also correct in Edge and IE 11. In Firefox, however, the following problems still occur with the example https://www.w3.org/WAI/tutorials/tables/irregular/#table-with-headers-spanning-multiple-rows-or-columns:
* JAWS jumps between columns when Ctrl+Alt+Down Arrow is used to navigate
* The column number is not displayed correctly on the Braille display (column 2 as column 1 etc.)
* The column headings are not displayed correctly (offset by one column each)

The reason for the problem is that rowspan is misinterpreted. All cells affected by rowspan are treated as if they were not present

I consider this to be a serious problem, which should be solved as soon as possible

**Comment by carmacleod** [comment]
The [Example 1 table](https://www.w3.org/WAI/tutorials/tables/irregular/#table-with-two-tier-headers) also shows the problem in Firefox (in case an example with column spans is useful for debugging).
- JAWS jumps between columns when Ctrl + Alt + Down Arrow is used to navigate
- move the VC to 80,000 (2nd column, 5th row) and try to use Ctrl + Alt + Up Arrow and JAWS says "Top of Column", but it should move to the "Sold" and "Venus" headers
- move the VC to 50,000 (2nd column, 3rd row) and type Insert + Shift + End (Read from Top of Column) and JAWS says "Mars, Sold, 50,000", but it should say "Mars, Produced, 50,000".

The problem does not happen in Chrome. It also does not happen with NVDA+Firefox or NVDA+Chrome.

**Comment by stevefaulkner** [comment]
Works for me with Chrome and JAWS 2024.2310.70

**Comment by JAWS-test** [comment]
@stevefaulkner I would like to ask you to reopen the ticket. The ticket mentions problems with Chrome and Firefox. The problems in Chrome have indeed been solved, but unfortunately not in Firefox. If you go to https://www.w3.org/WAI/tutorials/tables/irregular/#table-with-two-tier-headers with Firefox, you will notice that JAWS jumps wildly between the columns when navigating the table ALT+CTRL+arrow up or down or says in the middle of the table that the beginning of the table has been reached

**Comment by daniel-montalvo** [comment]
Please reopen. This seems broken in 2025 as well.
